### PR TITLE
Close #18: Added XO choose and fixed some alpha-beta problem.

### LINKED ISSUE #18: Add support XO choose

### PR DESCRIPTION
**Pull request by ognick**

### CODE REVIEW

File: core/src/main/scala/RunSolver.scala [hunk 1]
```diff
@@ -5,14 +5,14 @@ import wop.game.ai.WopSolver
 import scala.annotation.tailrec
 
 object RunSolver extends App {
-  @tailrec def loop(state: WopState): Option[WopState] = {
+  @tailrec def loop(state: WopState, whoAI: Player): Option[WopState] = {
     println(state)
```
Comment (fomkin) [inline core/src/main/scala/RunSolver.scala]: Remove this

File: core/src/main/scala/RunSolver.scala [hunk 2]
```diff
@@ -5,14 +5,14 @@ import wop.game.ai.WopSolver
...
     state match {
-      case s: InProgress if s.player == Player.P2 =>
+      case s: InProgress if s.player == whoAI =>
         WopSolver.minMax(state) match {
           case Some(p) => {
             s(p) match {
-              case Right(newState) => loop(newState)
+              case Right(newState) => loop(newState, whoAI)
               case x => {
                 println("i dont know", x)
```
Comment (fomkin) [inline core/src/main/scala/RunSolver.scala]: Remove this.

File: core/src/main/scala/RunSolver.scala [hunk 3]
```diff
@@ -28,24 +28,31 @@ object RunSolver extends App {
         println("input local x,y")
```
Comment (fomkin) [inline core/src/main/scala/RunSolver.scala]: Remove this

File: core/src/main/scala/RunSolver.scala [hunk 4]
```diff
@@ -28,24 +28,31 @@ object RunSolver extends App {
...
         prevState(point) match {
-          case Right(newState) => loop(newState)
+          case Right(newState) => loop(newState, whoAI)
           case _ =>
             println("wrong input")
-            loop(prevState)
+            loop(prevState, whoAI)
         }
       case prevState: WopState.Select =>
         println("input global x,y")
```
Comment (fomkin) [inline core/src/main/scala/RunSolver.scala]: Remove this

File: core/src/main/scala/wop/game/ai/WopSolver.scala [hunk 5]
```diff
@@ -133,16 +132,16 @@ object WopSolver {
...
                       val bestEval = alphaBeta.compare(currEval, eval)
                       if (alphaBeta.testPruning(bestEval._1)) bestEval
-                      else processChild(tl, player, bestEval)
+                      else processChild(tl, bestEval)
                     case _ => currEval
-
                   }
 
                 val initEval = (alphaBeta.initial, currPoint)
+                println(alphaBeta.step, "----Eval", alphaBeta.initial, "depth", depth)
```
Comment (fomkin) [inline core/src/main/scala/wop/game/ai/WopSolver.scala]: Remove this

**Review by fomkin — CHANGES_REQUESTED** [review]
Remove printlns and thats all.

File: core/src/main/scala/RunSolver.scala (hunk 2 above)
Comment (ognick) [inline core/src/main/scala/RunSolver.scala]: it is UI message

File: core/src/main/scala/wop/game/ai/WopSolver.scala (hunk 5 above)
Comment (ognick) [reply core/src/main/scala/wop/game/ai/WopSolver.scala]: ok

**Review by fomkin — APPROVED** [review]

**Pull request merged**